lightningd/peer_htlcs.c

```diff
@@ -1848,9 +2035,13 @@ void htlcs_reconnect(struct lightningd *ld,
 	for (hin = htlc_in_map_first(&unprocessed, &ini); hin;
 	     hin = htlc_in_map_next(&unprocessed, &ini)) {
 		log_unusual(hin->key.channel->log,
-			    "Failing old unprocessed HTLC #%"PRIu64,
+			    "Replaying old unprocessed HTLC #%"PRIu64,
 			    hin->key.id);
-		fail_in_htlc(hin, WIRE_TEMPORARY_NODE_FAILURE, NULL, NULL);
+		if (!peer_accepted_htlc(hin->key.channel, hin->key.id, true, &failcode)) {
+			fail_in_htlc(hin, WIRE_TEMPORARY_NODE_FAILURE, NULL, NULL);
```

Comment:
hi, cdecker, when `parse_onionpacket()` filed it may set a `failcode` and return `op` as `NULL,` then `peer_accepted_htlc()` whould return false

> `	op = parse_onionpacket(tmpctx, hin->onion_routing_packet,
> 			       sizeof(hin->onion_routing_packet),
> 			       failcode);`

should there be `fail_in_htlc(hin, failcode? failcode : WIRE_TEMPORARY_NODE_FAILURE, NULL, NULL);`

I aslo have a related question...if there need to store forward status as `LOCAL_FAILED`? 
It seems I missed this case...

Reply:
Indeed failcode is not always set; sometimes it will try to fail the channel itself.  I don't know what happens in those cases for this code path, though I suspect it Can't Happen?

As to LOCAL_FAILED, I think if we crash at this point we'll just replay anyway, so it's not vital.  But it would be a good idea, I think.

Reply:
> Indeed failcode is not always set; sometimes it will try to fail the channel itself. I don't know what happens in those cases for this code path, though I suspect it Can't Happen?

In those cases the HTLC will be marked as failed in the DB in the same transaction it was added in, hence it won't be replayed. So yes, this should never happen :tm: 

> As to LOCAL_FAILED, I think if we crash at this point we'll just replay anyway, so it's not vital. But it would be a good idea, I think.

 Indeed, that is the reason the documentation explicitly states that the plugin needs to implement the hook method in an idempotent manner.